src/main/java/baritone/behavior/MemoryBehavior.java

```diff
@@ -120,24 +120,24 @@ public synchronized void onReceivePacket(PacketEvent event) {
 
     @Override
     public void onBlockInteract(BlockInteractEvent event) {
-        if (event.getType() == BlockInteractEvent.Type.USE && BlockStateInterface.getBlock(event.getPos()) instanceof BlockBed) {
+        if (event.getType() == BlockInteractEvent.Type.USE && BlockStateInterface.getBlock(ctx, event.getPos()) instanceof BlockBed) {
             baritone.getWorldProvider().getCurrentWorld().getWaypoints().addWaypoint(new Waypoint("bed", Waypoint.Tag.BED, event.getPos()));
         }
     }
 
     @Override
     public void onPlayerDeath() {
-        baritone.getWorldProvider().getCurrentWorld().getWaypoints().addWaypoint(new Waypoint("death", Waypoint.Tag.DEATH, playerFeet()));
+        baritone.getWorldProvider().getCurrentWorld().getWaypoints().addWaypoint(new Waypoint("death", Waypoint.Tag.DEATH, ctx.playerFeet()));
     }
 
     private Optional<RememberedInventory> getInventoryFromWindow(int windowId) {
         return this.getCurrentContainer().rememberedInventories.values().stream().filter(i -> i.windowId == windowId).findFirst();
     }
 
     private void updateInventory() {
-        getInventoryFromWindow(player().openContainer.windowId).ifPresent(inventory -> {
+        getInventoryFromWindow(ctx.player().openContainer.windowId).ifPresent(inventory -> {
             inventory.items.clear();
-            inventory.items.addAll(player().openContainer.getInventory().subList(0, inventory.size));
+            inventory.items.addAll(ctx.player().openContainer.getInventory().subList(0, inventory.size));
         });
     }
```

Comment:
General on MemoryBehavior: could we instead hook into where the packets are processed on the main thread, instead of where netty receives them? So that we avoid any possible concurrent modification exceptions?

Reply:
Hmmmm. Are you sure? 0x22 was getting the h00000 error in MemoryBehavior, which is caused when it's called from a different thread. Psure the current hook is on a different thread. I'm not referring to context, just to possible ConcurrentModificationExceptions in the remembered chest contents.

Reply:
Yes, by packet processing I meant the handling of "what the packet does". The packet hook that we use in MemoryBehavior is called from the network handler thread, before the packet is actually processed in the main thread. The code in the game is really cancer, but basically it makes an attempt to process packets in the handling thread, and then an exception is thrown terminating the thread after it has been queued to be processed in the main thread.

Reply:
OK i guess we just need to be really careful synchronizing this

Reply:
Yep.